### PR TITLE
Whisper quantized wasm

### PR DESCRIPTION
**Pull request by radames**
* add support for the quantized wasm
* remove the model.rs and pointed to transformers/whisper.rs
* tested trunk demo and move assets around 
* send o a PR here https://huggingface.co/spaces/lmz/candle-whisper

It will be nice to adapt the `convert_and_run` in the future to send segments as it decode its, something like `next_segment`

### CODE REVIEW

File: candle-wasm-examples/whisper/src/worker.rs [hunk 1]
```diff
@@ -247,17 +300,44 @@ impl Decoder {
...
+        let task = match md.task {
+            Some(task_str) => {
+                let task_string = task_str.to_owned();
+                match task_string.as_str() {
+                    "translate" => Some(Task::Translate),
+                    _ => Some(Task::Transcribe),
+                }
+            }
+            _ => Some(Task::Transcribe),
+        };
```
Comment (radames) [inline candle-wasm-examples/whisper/src/worker.rs:330]: is there a better way to to this? I want to send a `string` or None,  also tried to use an Enum with wasm_bindgen

**Comment by LaurentMazare** [comment]
Oh that's so nice that you removed this model duplication, thanks!

**Comment by LaurentMazare** [comment]
Could you run rustfmt? Seems like it's picking some small diff in the CI.

File: candle-wasm-examples/whisper/src/worker.rs (hunk 1 above)
Comment (LaurentMazare) [reply candle-wasm-examples/whisper/src/worker.rs:330]: Maybe you could have a look at [this example](https://github.com/huggingface/candle/blob/11d3687cc655f8f79d856342a5539a9274e96df4/candle-examples/examples/quantized/main.rs#L306-L311), with the same`as_deref` thing you can probably do it with a single statement.

**Comment by radames** [comment]
thanks @LaurentMazare, much better with `as_deref`.
I did run `cargo fmt  --all -- --check` now does it fix the issue?

**Review by LaurentMazare — APPROVED** [review]

**Comment by LaurentMazare** [comment]
Looks great, thanks!

**Pull request merged**